src/quo/theme.cljs

```diff
@@ -1,16 +1,27 @@
 (ns quo.theme
   (:require
     ["react" :as react]
+    [oops.core :as oops]
     [react-native.core :as rn]))
 
-(defonce ^:private theme-context (react/createContext :light))
+(defonce ^:private context (react/createContext nil))
 
 (defn provider
-  [theme & children]
-  (into [:> (.-Provider theme-context) {:value theme}]
+  [data & children]
+  (into [:> (.-Provider context) {:value data}]
         children))
 
 (defn use-theme
   "A hook that returns the current theme keyword."
   []
-  (keyword (rn/use-context theme-context)))
+  (if-let [data (rn/use-context context)]
+    (keyword (oops/oget data :theme))
+    :light))
```

Comment:
could you elaborate on why condition was added here

Reply:
Hi, for the screens, we are using the `quo.theme/provider` wrapper and providing the theme, which is then used in components through `quo.theme/use-theme`. However, for component tests, we're applying this only to a subset of components (via `render-with-theme-provider`), while others simply call `render`, but still expect values from `quo.theme`. Additionally, it felt safer to include a null check with a fallback.